### PR TITLE
Various TypeScript fixes across several packages

### PR DESCRIPTION
**Pull request by wyze**
These are some of the fixes we still had locally after upgrading to the latest version of the charts we use.

If you review each commit, they should be descriptive enough, but if not, I can expand further.

### CODE REVIEW

File: packages/legends/index.d.ts [hunk 1]
```diff
@@ -93,4 +94,6 @@ declare module '@nivo/legends' {
     }
 
     export type QuantileLegendSvg = React.FunctionComponent<QuantileLegendProps>
+
+    export const BoxLegendSvg: React.FC<LegendProps & Dimensions>
```
Comment (agustingabiola) [inline packages/legends/index.d.ts]: Looking into `https://github.com/plouc/nivo/blob/master/packages/legends/src/svg/BoxLegendSvg.js` Proptypes I don't think these match.

Comment (wyze) [reply packages/legends/index.d.ts]: Which ones specifically?

Comment (agustingabiola) [reply packages/legends/index.d.ts]: https://github.com/plouc/nivo/blob/master/packages/legends/src/svg/BoxLegendSvg.js#L113

You don't pass `height` & `width` but `containerWidth` & `containerHeight`

**Comment by mahatosourav91** [comment]
@wyze @agustingabiola @plouc 

Would be possible to get this merged and have a minor release of legends?

File: packages/legends/index.d.ts (hunk 1 above)
Comment (wyze) [reply packages/legends/index.d.ts]: Good catch, thank you.

**Pull request merged**